### ISSUE TITLE
[Bug]: preferDesktop option crashes the dapp

### ISSUE THREAD

**Issue by christopherferreira9** [MetaMask/metamask-sdk#790]
### SDK

Web

### Provide environment information

-

### MetaMask SDK Version

0.18.1

### MetaMask Mobile app Version

-

### Describe the Bug

Setting the option `preferDesktop` to true is currently crashing the dapp with an exception and throwing the user to the Extension web store.

Exceptions being thrown:
```
TypeError: Cannot assign to read only property 'ethereum' of object '#<Window>'
    at new ID (http://localhost:3000/static/js/bundle.js:28550:99)
    at ID.init (http://localhost:3000/static/js/bundle.js:28590:86)
    at BD 
```

```
Ethereum instance not intiialized - call Ethereum.factory first.
    at ID.getProvider (http://localhost:3000/static/js/bundle.js:28602:88)
```

### Expected Behavior

The dapp presents a dialog to choose between extension or mobile with the extension being the first to be displayed.

### To Reproduce

Set the option preferDesktop to true and try to connect a dapp

**Comment by tiagofilipenunes** [comment]
Same issue here, same option used and same error.

Using wagmi 2.10.9, initializing metaMask as 

```typescript
import { metaMask } from 'wagmi/connectors';

metaMask({
  extensionOnly: false,
  preferDesktop: true,
  dappMetadata: {
    name: config.appName,
    url: config.appUrl,
  },
});
```

Getting the same error on Sentry:
```
Cannot assign to read only property 'ethereum' of object '#<Window>'
```

Here's a working example:
https://stackblitz.com/edit/vitejs-vite-ajga6b

**Comment by christopherferreira9** [comment]
Fixed as of 0.27.0. The fix on the wagmi side is currently under test @tiagofilipenunes
